components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -112,7 +112,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
 
         if (StringUtils.isEmpty(userStoreDomain)) {
             context.setSubject(
-                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserName));
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(fullyQualifiedName));
```

Comment:
I think as the first step we can do this fix for federated authenticator mode. WDYT @gasform ?